Template/LO/SubProcesses/cuts.f

```diff
@@ -256,11 +256,15 @@ LOGICAL FUNCTION PASSCUTS(P)
 c               call set_ren_scale(P,scale)
 c            endif
 c         endif
-         
+
+c     If scale is fixed, update G-dependent couplings for VECSIZE_MEMMAX events
+c     This is called only once in the application (FIRSTTIME=.true.)
+c     Only VECSIZE_USED events are needed, but this variable is unknown here
+c     Using VECSIZE_MEMMAX is correct and has a negligible performance overhead
 
          if(fixed_ren_scale) then
             G = SQRT(4d0*PI*ALPHAS(scale))
-            do i =1, VECSIZE_USED
+            do i =1, VECSIZE_MEMMAX ! no need to use VECSIZE_USED here
                call update_as_param(i)

```

Comment:
Hi, if I remember correctly... you are completely correct that VECSIZE_USED would be more appropriate here. The problem is that, if we have this as an argument to pass from function to function and not as a common, then I had found it difficult to pass it here. But I can try again if you want.

This is in the passcut function, which is called in two places only IIUC

1. In dsigproc (auto_dsig.f), where however I see
```
      DSIGPROC=0D0
C     not needed anymore ... can be removed ... set for debugging only
C        
      IF (.NOT.PASSCUTS(P1)) THEN
        STOP 1
      ENDIF
``` 
Maybe this can be removed?...

2. In sample_full from dsample.f, where indeed vecsize_used is available, so I can add that trivially

3.  In addition there are two more calls to passcuts, both commented out
- one in dsigproc_vec from auto_dsig.f
- one in gen_mom from genps.f

Let's say: if you confirm that the two commented out lines can be removed (or kepyt commented out), and that also the dsig call can be commented out, then I can add vecsize_used to the passcuts function signature as an argument, and use it instead of vecsize_memmax. Should I do that?

Otherwise, I do not think that you get a segfault, just because in practice update_as_param(i) and the coup2(i) that it calls internally are both setting a variable(i), with no input variable at all. The memory is allocated up to vecsize_memmax, so you just end up creating a vector that is too long, longer than is needed, but the performance penalty is really negligible as I think this is done once, not for each event. Ugly, but it would work.

What do you prefer, should I change the passcuts signature adding vecsize_used and comment out passcuts in sigproc?

Reply:
Urgh now I see another thing that I did not like. Yes I can pass VECSIZE_USED to passcuts, but this is really ugly because actually it is only used once. 

The problem is that the first call to passcuts is where the couplings are set (depending on scale)! This is not particularly nice. It would probably be better to do it outside passcuts?

Reply:
One think that is also ugly is that I would anyway need to set one variable as a common block since 
VECSIZE_USED  needs to be part of the run_card.dat and that those parameter are for most of them --if not all--
have their definition in run.inc and their value in run_card.inc  which is include/evaluated during the initialization.

So for the moment, I was thinking to set for the production code
VECSIZE_USED_USER part of the common block
and ensure that in production
VECSIZE_USED_USER = VECSIZE_USED 
and ensure that
VECSIZE_MEMMAX>= VECSIZE_USED_USER 
(so force a recompilation if that one is increase but do not recompile if it is decrease)

The advantages of having VECSIZE_USED_USER different of  VECSIZE_USED is that you can have as much hack as you need within the code and have that value changed at anytime, which I would not like for VECSIZE_USED_USER.
But maybe you do not need that.

For the passcuts, I agree that that one is quite ugly to have. If this is for initialization maybe using VECSIZE_USED_USER is ok. For the pascuts, I would say that this is important to be able to call it from anywhere in the code.

But indeed, it might be a good idea to move that part of the code outside of the cuts function where it does not make that much sense. ( will look for that in "my" other work in progress branch)